### DISCUSSION TITLE
Tasmota and "Matter Protocol" Trying to make sense of what it means.

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#18540] [Q&A] [answered]
I have been doing some reading on "Matter Protocol" and so far it is my understanding that Matter Protocol will make it possible for home automation products from various manufacturers to work together using only a compatible border router .

I am Still trying to understand what a "Matter Protocol enabled Tasmota" will mean for me the casual user of Home Automation devices. Let us say I have a Mater enabled google device, matter enabled Apple Device, along with matter enabled Zigbee device, Matter enabled alexa device, and Matter enabled Tasmota device, now what am I looking at? 

In addition I  have a "tasmota enabled Sonoff ZigBee coordinator" now do I need this coordinator or will I have to flash and make it a "Tasmota flashed border router/bridge" to make use of it? Or will I not need it at all and everything listed above will work together seamlessly since everything is matter enabled and communicating over the home wifi router IP network?

**Comment by s-hadinger** [comment, marked as answer]
It's still early days. Currently it means that you can control directly an esp32 based Tasmota device from Alexa/Apple/Google gateways without the need of an additional component like RPi nor any additional skill or equivalent. 
You can expect direct Matter to Zigbee translators in the future, if not Tasmota should be able to evolve in this direction as well.

**Reply by NsinghP** [reply]
> ...Currently it means that you can control directly an esp32 based Tasmota device from Alexa/Apple/Google gateways ....

Ok, so that rules out any ESP8266 based devices to be part of the "Matter" equation?

**Reply by s-hadinger** [reply]
Yes. But I'm exploring if a single esp32 could act as a bridge for esp8266 devices

**Reply by NsinghP** [reply]
> Yes. But I'm exploring if a single esp32 could act as a bridge for esp8266 devices

Alright, that explains your comment about  "I will start exploring Bridge mode...." in discussion https://github.com/arendst/Tasmota/discussions/17872 I think now I have better understanding of where the Tasmota flashed Sonoff Zigbee BridgePro fits in. Thanks. And if I am not mistaken then Alexa/Apple/Google gateways will act like border routers?

**Reply by s-hadinger** [reply]
I'm not exactly sure if you can call them border routers. But they ensure the commissioning in the Matter fabric (that you can see as a Matter community or network). The big new feature compared to zigbee is that a single device can be part of multiple fabrics and be controlled at the same time by an AppleTV and an Alexa Echo device.

**Reply by NsinghP** [reply]
> I'm not exactly sure if you can call them border routers. But they ensure the commissioning in the Matter fabric (that you can see as a Matter community or network). The big new feature compared to zigbee is that a single device can be part of multiple fabrics and be controlled at the same time by an AppleTV and an Alexa Echo device.

Ok, so after successful implementation of Bridge Mode on ESP32 device, ESP8266 devices will  become part of this fabric and will not need Hue Emulation enabled. Thats one less step in ESP8266 device configuration and good news!

**Reply by s-hadinger** [reply]
That's the plan. But there are still many challenges to solve

**Reply by NsinghP** [reply]
Today I procured an "ESP32 38 Pin Development Board" and would love to flash it with bleeding edge "Matter enabled Tasmota bridge firmware" and test it with more than 10 ESP8266 based devices that I have flashed with Tasmota and currently control with Alexa using Hue Emulation. Let me know if I can be of any assistance in overcoming the challenges in development of "Bridge Mode Tasmota" firmware.

**Reply by s-hadinger** [reply]
Thanks for this proposal. Preliminary tests show that there are not so much challenges. I need time to implement it. Stay tuned

**Reply by Bibabutzemann123** [reply]
I thought its already possible to control Tasmota devices directly with an Echo. Whats the difference between that and using Matter?

**Reply by sfromis** [reply]
The difference is the Matter ecosystem, instead of the walls of the Amazon ecosystem. Of course, it is still too early to tell how widespread Matter usage is going to be, but major players are part of the alliance.

**Reply by Bibabutzemann123** [reply]
But with both methods there is only a local connection with the the Echo without any need of additional Hardware, right?
The advantage going this route is that you dont need an Matter-ready module like the ESP32.

But yeah i see the advantage doing it with Matter.
For example with google Nest there is no way around Matter, if you wanna connect directly with a Tasmota device.

**Reply by s-hadinger** [reply]
Yes indeed. But you still need a local gateway. The smartphone alone is not enough. For some platforms, when your smartphone is on the same LAN, there is a direct communication to ensure low latency

**Reply by Bibabutzemann123** [reply]
Right, but something like the google nest mini will do.
I think thats already a great solution for people like me, who only want to control few switches via google home but dont want to be dependant on several cloud servers.

**Reply by s-hadinger** [reply]
That's exactly the reason why I took the effort of implementing Matter on Tasmota!

**Reply by Bibabutzemann123** [reply]
Oh that was you :D
Thank you for your effort!
